ostree-backup.sh

```diff
@@ -35,13 +48,26 @@ tar czf /var/tmp/backup/var.tgz \
     --exclude='/var/lib/cni/bin/*' \
     --selinux \
     /var
-ostree admin config-diff | awk '{print "/etc/" $2}' | xargs tar czf /var/tmp/backup/etc.tgz
+ostree admin config-diff | awk '{print "/etc/" $2}' | xargs tar czf /var/tmp/backup/etc.tgz --selinux
 rpm-ostree status > /var/tmp/backup/rpm-ostree.status
-ostree commit --branch backup /var/tmp/backup
+rpm-ostree status -v --json > /var/tmp/backup/rpm-ostree.json
+cp /etc/machine-config-daemon/currentconfig /var/tmp/backup/mco-currentconfig.json
+ostree commit --branch $backup_tag /var/tmp/backup
 
 # Create credentials for pushing the generated image
 mkdir /root/.docker
 cp backup-secret.json /root/.docker/config.json
 
 log_it Encapsulating and pushing backup OCI
-ostree container encapsulate backup registry:$backup_refspec --repo /ostree/repo --label ostree.bootable=true
+ostree container encapsulate $backup_tag registry:$backup_refspec --repo /ostree/repo --label ostree.bootable=true
+
+log_it Encapsulating and pushing base OCI
+base_commit=$(rpm-ostree status -v --json | jq -r '.deployments[] | select(.booted == true).checksum')
+ostree container encapsulate $base_commit registry:$base_refspec --repo /ostree/repo --label ostree.bootable=true
+
+# If there's a parent to that commit, also encapsulate it
```

Comment:
So if we're always using the original base container image, then what we want to do here is change things so that if `base-checksum != null`, then we also encapsulate the derived AKA `checksum`.

Reply:
So, the idea is:

`if base-checksum == null`, then we `do not encapsulate anything`, because we use the pristine RHCOS image, and in destination host deploy that RHCOS image

`if base-checksum != null`, then we `encapsulate the checksum`, and in the destination host we deploy that OCI image with `checksum`, and unencapsulate the pristine RHCOS image so we have the `base-checksum` also in our ostree repo

That's the workflow I thought at the first time, but I find a weird behaviour, and this is, the pristine RHCOS image gets imported with a different checksum, and thus rpm-ostree breaks because it cannot find the `base-checksum`, however, if I do encapsulate that `base-checksum` when creating the backup, when importing it does have the proper checksum ... so I dont know what's going on:

Check values in rpm-ostree status that we imported
```
[root@sno core]# ostree cat backup /rpm-ostree.json | jq -r '.deployments[] | select(.booted == true) | { "checksum": .checksum, "base-checksum": ."base-checksum", "container-image-reference": ."container-image-reference" }'
{
  "checksum": "5f80ce35c586bbf8cce9b68616d17d2f33907352317743571b3368b7ebbf1947",
  "base-checksum": "906a432bef54cda8973c784a09449e7ca82185dfb6b8f686f678ffe77846f258",
  "container-image-reference": "ostree-unverified-registry:quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:64e718da9e357a00954fbb52d0aebd121ba8352812461b65ca2380d92dd50a8c"
}
```

Import container-image-reference OCI to see which checksum it creates:
```
[root@sno core]# ostree container unencapsulate --repo /ostree/repo --write-ref basechecksum ostree-unverified-registry:quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:64e718da9e357a00954fbb52d0aebd121ba8352812461b65ca2380d92dd50a8c
   Downloading...                                                                                                                                                                              
Imported: basechecksum => 43254a36e7ac85bf3a1804a087a42f6788763281856bba356520424b0269b737
```
We can see that the checksum does not match the base-checksum from rpm-ostree status

Now I will import the backup I made with `ostree container admin encapsulate`
```
[root@sno core]# ostree container unencapsulate --repo /ostree/repo --write-ref basechecksum2 ostree-unverified-registry:quay.io/jpolo/ostmagic:parent
   Downloading...                                                                                                                                                                              
Imported: basechecksum2 => 906a432bef54cda8973c784a09449e7ca82185dfb6b8f686f678ffe77846f258
```
And we can see that this checksum DOES match the base-checksum ....

There is something I'm missing, but I dont know what ....